### ISSUE TITLE
link axes between two plots

### ISSUE THREAD

**Issue by swharden** [swharden/QuickPlot#11]
I want to be able to lock the axes of two subplots together. I think the best way to do this is like [how matplotlib does it](https://matplotlib.org/3.1.1/gallery/subplots_axes_and_figures/shared_axis_demo.html)

- [x] make Subplot() return the plot that was just created
- [x] create `plot.ShareX()` and `plot.shareY()`
- [x] move all mouse code out from QuickPlot and into the user controls
- [x] make sharex and sharey use the same axis class objects
- [x] ensure Render() knows to redraw axis-linked plots when told to only update a single plot

Example usage:
```cs
var plotA = figure.Subplot(2, 1, 1);
var plotB = figure.Subplot(2, 1, 1, sharex = plotA);
```

Links should be bidirectional: 
If I click-and-drag plot A, I want plot B to move with it. 
If I click-and-drag plot B, I want plot A to move with it.

**Comment by swharden** [comment]
I was so close! This is difficult because I can't copy axes between graphs because the mousetracker stores axes offsets and only actually modifies the axes after a "MouseUp". I want the effect to be live as I click-drag, so I might have to redesign how the mousetracker interacts with the axes 🤔

**Comment by StendProg** [comment]
What would you want to achieve with this links?
1. Linked axises have equal ranges ( min value, max value) and always in sync.
2. Linked axises have different ranges but recieved equal pan/zoom command raised from interaction.

If 2 you can make fake mouse call to linked plots. I think it easy to implement.

If you want to make this task more general. Not only 2 subplots may be connected. In this case I think better to store connections in special structure in figure class.

**Comment by swharden** [comment]
I'd like to do what matplotlib does in Python

```cs
import matplotlib.pyplot as plt
import numpy as np
xs = np.arange(100)/10
plt.figure()
ax1 = plt.subplot(211)
plt.plot(xs, np.sin(xs))
ax2 = plt.subplot(212, sharex=ax1)
plt.plot(xs, np.cos(xs))
plt.show()
```

![sharex](https://user-images.githubusercontent.com/4165489/67045108-a3f96980-f0fb-11e9-81a6-7e68c0bf8c0b.gif)

**Comment by StendProg** [comment]
https://github.com/swharden/QuickPlot/blob/1772d0fe34df15311f8ee77f461cb3b1302de02d/src/QuickPlot/Plot.cs#L99
```c#
var sharexAxes = sharex?.AxesAfterMouse();
var shareyAxes = sharey?.AxesAfterMouse();
```
Mix 3 of them to 1 in right way.
Just need additional know interaction goes on this plot or other to make correct mix.
Active subplot render in standart way. After calls render of shared subplot.  Which replace in render axes with not null (sharexAxes, shareyAxes).

**Comment by swharden** [comment]
@StendProg I liked your suggested solution, but it got complicated when several plots were involved. The mouse movement is what kept making this task difficult. 

I came up with a different solution. I moved all mouse code to the user controls (where it probably belongs anyway). Now I can use the same axis objects in multiple plots.

https://github.com/swharden/QuickPlot/blob/85ca379b1947bcf322e4fd67e48a380f6fd7d136/src/QuickPlot/Plot.cs#L61-L68

![shared2](https://user-images.githubusercontent.com/4165489/67058992-36fac980-f125-11e9-9a53-0a5d73155964.gif)
